### ISSUE TITLE
[HOLD for payment 2025-01-24] [$250]  IOS - Empty screen seen when navigating back from workspace description the second time

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#53589]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.71-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** nathanmulugetatesting+2338@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open IOS hybrid app
2. Create a workspace give it a custom description
3. Navigate to the inbox
4. Open workspace chat
5. Tap on the custom description to navigate to description setting of the workspace
6. Navigate back twice by tapping on the top left back icon until you reach workspace editor page
7. Tap on Workspace chat
8. Re do step 5
9. Tap on the top corner back icon or swipe from left to right

## Expected Result:
Empty screen does not get shown

## Actual Result:
Empty screen gets shown and the workspace profile settings page animation is wrong, it slides from right to left after the empty screen has been shown first

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/46ddf493-9168-4d87-987e-b90b5692ea02

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021865096772528155887</li>
        <li>Upwork Job ID: 1865096772528155887</li>
        <li>Last Price Increase: 2025-01-10</li>
<li>Automatic offers: </li>
<ul>
<li>ishpaul777 | Reviewer | 105677024</li>
<li>ikevin127 | Contributor | 105677027</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @muttmuure</details>

**Comment by ikevin127** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue

Empty screen gets shown and the workspace profile settings page animation is wrong, it slides from right to left after the empty screen has been shown first.

### What is the root cause of that problem?

**TLDR**: The same RCA detailed in my [proposal](https://github.com/Expensify/App/issues/53600#issuecomment-2524799994) of this other [issue](https://github.com/Expensify/App/issues/53600).

> [!note]
> This issue is only present on narrow layout devices (native / mWeb) where screens displayed in RHP act as a modal.

https://github.com/Expensify/App/blob/3baf965618ced48845ddd7534b0fad4d4110b5de/src/pages/workspace/WorkspaceInitialPage.tsx#L360-L376

When we open the workspace's initial page, the active route becomes `Workspace_Initial`. However, the current `canAccessRoute` logic only checks for the sub-page route - therefore when it compares the active route with itself, it fails because `menuItems` doesn't contain `Workspace_Initial`.

We actually have the logic to only run it once [here](https://github.com/Expensify/App/blob/3baf965618ced48845ddd7534b0fad4d4110b5de/src/pages/workspace/WorkspaceInitialPage.tsx#L368-L371), but when we open the search page, all routes are "hidden" as explained [here](https://github.com/Expensify/App/blob/3baf965618ced48845ddd7534b0fad4d4110b5de/src/libs/Navigation/AppNavigator/createResponsiveStackNavigator/useStateWithSearch.ts#L44-L57). Because of this, the `wasRendered` logic in `WorkspaceInitialPage` won't work as expected -> when we try to open the filter page, the previously "hidden" routes are added back to the stack and `canAccessRoute` becomes `false` which will dismiss any RHP that we try to open (Filters).

Credits to **bernhardoj** for the RCA.

### What changes do you think we should make in order to solve the problem?

We had a similar issue which I reviewed recently [here](https://github.com/Expensify/App/issues/52258) where we navigated away from `WorkspaceProfileSharePage` to a report **without dismissing the currently opened modal** which caused a weird UI bug in the QR Code because the same modal would be opened twice.

Same as proposed in the other [issue](https://github.com/Expensify/App/issues/53600#issuecomment-2524799994), a practical solution for this kind of problem is to simply replace the way we navigate away from WS to the report here:

https://github.com/Expensify/App/blob/b595f5d24141a403fcc3f44c73d00cd6951dca7d/src/pages/workspace/WorkspaceInitialPage.tsx#L457

to use `Navigation.dismissModal` instead:

```ts
onPress={() => Navigation.dismissModal(currentUserPolicyExpenseChat?.reportID ?? '-1')}
```

**Why dismissing the modal first fixes our issue ?**
Because as explained in RCA, on narrow layout devices (native / mWeb) screens displayed in RHP act as a modal and by dismissing it when navigating to the report, all the conditions which lead to the empty screen showing when navigating back in our case, will now work as expected.

> [!important]
> I proposed the same solution for [this issue](https://github.com/Expensify/App/issues/53600) and because this solution would fix both issues, if my solution is selected on any of the two issues -> we should put the other one on **HOLD**.

### Result video (before / after)

<details>
<summary>iOS: Native</summary>

| Before | After |
| ------------- | ------------- |
| <video src="https://github.com/user-attachments/assets/713ac42c-55b9-48dc-8691-ddcea7890289"> | <video src="https://github.com/user-attachments/assets/e3ae4bfd-5523-47bd-9d6d-f78a05d16b87"> |

</details>

**Comment by ikevin127** [comment]
> Will this issue be fixed in https://github.com/Expensify/App/issues/53600, i see we got a proposal selected there?

cc @bernhardoj for input on this question

**Comment by bernhardoj** [comment]
No

**Comment by ikevin127** [comment]
**Expected result** states:

> Empty screen does not get shown

From you video it looks like that the expected result is fulfilled with the proposed solution since the expected result did not specify what page we should land on. But ok, I'll look into this and try to come up with an alternative solution that would fulfill your expectations 👍

**Comment by ikevin127** [comment]
I performed a `npm ci` and **iOS: Native** app rebuild and I can't reproduce the issue anymore.

https://github.com/user-attachments/assets/661187ac-04ee-4993-b064-e5a39592ccd0

**Comment by bernhardoj** [comment]
I think it was "solved" by https://github.com/Expensify/App/pull/53985, but it causes the navigation issue mentioned [here](https://github.com/Expensify/App/issues/53589#issuecomment-2546388769) : https://github.com/Expensify/App/issues/54303.

**Comment by shawnborton** [comment]
I think the video looks correct? It seems to be navigating sequentially based on the browsing history.

**Comment by shawnborton** [comment]
Hmm maybe I am just not understanding the video correctly. Why does it immediately navigate to the description page? That does feel weird.

**Comment by dannymcclain** [comment]
>Why does it immediately navigate to the description page? That does feel weird.
>That is because description is pressable and i clicked on it to navigate to the description page.

I agree this is weird. I've long felt that the description should not ever be navigable. You should have to go into the workspace profile to edit the description. I think it's super weird that you can click on it in the message body AND in the header. Both of those interactions feel really unexpected and wrong to me.

**Comment by shawnborton** [comment]
Ah I get it now. But even if we keep it so that you can navigate to the description RHP, why aren't we just launching it directly in the RHP rather than going to the workspace overview page first and then the description RHP?

**Comment by dannymcclain** [comment]
>should we make it not pressable and read only

This is my vote, but might be worth a quick slack convo in case other peeps are SUPER attached to that behavior. It just seems like it causes more harm than good to me, and the path to edit the description is pretty straight forward. So I don't super see what problem having all these things clickable is solving.

**Comment by dubielzyk-expensify** [comment]
Yeah I tend to agree with you Danny, but I might be missing some historical context for this

**Comment by ikevin127** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Remove the navigation to description page when pressing on the description text of a workspace with custom description.

### What is the root cause of that problem?

Root cause of the issue comes from here where if welcomeMessage contains messageHtml, which means we set custom description, UI will render a pressable which will navigate us to WS description page if pressed:

https://github.com/Expensify/App/blob/7aaeef3bc87e681351b3d94abcf8556a961591af/src/components/ReportWelcomeText.tsx#L107-L120

### What changes do you think we should make in order to solve the problem?

Replace the Pressable wrapping the RenderHTML component with a View and pass almost the same styles as the pressable had:

```ts
style={[styles.renderHTML, styles.cursorText]}
```

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by ikevin127** [comment]
Working on the PR right now!

**Comment by ikevin127** [comment]
PR https://github.com/Expensify/App/pull/55177 is ready for review! 🚀

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by ikevin127** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by ikevin127** [comment]
cc @muttmuure for payments (tagged on Slack [thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1737827931316559) as well)

**Comment by mallenexpensify** [comment]
Contributor: @ikevin127  paid $250 via Upwork
Contributor+: @ishpaul777 paid $250 via Upwork

 Test case - https://github.com/Expensify/Expensify/issues/465111

Think I got it all, thanks!
